Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -0,0 +1,902 @@
+            int const n_coefs_z = m_stencil_coefs_z.size();
+
+            // Extract tileboxes for which to loop
+            Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect());
+            Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
+            Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
+
+            // Loop over the cells and update the fields
+            amrex::ParallelFor(tbx, tby, tbz,
+
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                    if (coupling == 1){
+                        Hx(i, j, k) = Hx_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDz(Ey, coefs_z, n_coefs_z, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDy(Ez, coefs_y, n_coefs_y, i, j, k) 
+                                                      - M_xface(i, j, k, 0) + M_xface_old(i, j, k, 0);
+                    }
+                    else{
+                        Hx(i, j, k) = Hx_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDz(Ey, coefs_z, n_coefs_z, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDy(Ez, coefs_y, n_coefs_y, i, j, k);
+                    }
+                },
+
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                    if (coupling == 1){
+                        Hy(i, j, k) = Hy_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDx(Ez, coefs_x, n_coefs_x, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDz(Ex, coefs_z, n_coefs_z, i, j, k) 
+                                                      - M_yface(i, j, k, 1) + M_yface_old(i, j, k, 1);
+                    }
+                    else{
+                        Hy(i, j, k) = Hy_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDx(Ez, coefs_x, n_coefs_x, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDz(Ex, coefs_z, n_coefs_z, i, j, k);
+                    }
+                },
+
+                [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                    if (coupling == 1){
+                        Hz(i, j, k) = Hz_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDy(Ex, coefs_y, n_coefs_y, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDx(Ey, coefs_x, n_coefs_x, i, j, k) 
+                                                      - M_zface(i, j, k, 2) + M_zface_old(i, j, k, 2);
+                    }
+                    else{
+                        Hz(i, j, k) = Hz_old(i, j, k) + 1. / PhysConst::mu0 * dt * T_Algo::UpwardDy(Ex, coefs_y, n_coefs_y, i, j, k) - 1. / PhysConst::mu0 * dt * T_Algo::UpwardDx(Ey, coefs_x, n_coefs_x, i, j, k);
+                    }
+                }
+
+            );
+        }
+
+        // Check the error between Mfield and Mfield_prev and decide whether another iteration is needed
+        amrex::Real M_iter_maxerror = -1.0;
+        for (int i = 0; i < 3; i++){
+            for (int j = 0; j < 3; j++){
+                Real M_iter_error = Mfield_error[i]->norm0(j);
```

Comment:
```suggestion
        for (int iface = 0; iface < 3; iface++){
            for (int jcomp = 0; jcomp < 3; jcomp++){
                Real M_iter_error = Mfield_error[iface]->norm0(j);
```

Reply:
fix the part that says norm0(j) to norm0(jcomp)

Reply:
Thank you so much!